mhs/common/mhs_common/state/tests/test_work_description.py

```diff
@@ -127,15 +129,85 @@ def test_constructor(self):
         work_description = wd.WorkDescription(persistence, input_data)
 
         new_data = copy.deepcopy(input_data)
-        new_data[wd.DATA][wd.STATUS] = wd.MessageStatus.OUTBOUND_MESSAGE_ACKD
+        new_data[wd.DATA][wd.OUTBOUND_STATUS] = wd.MessageStatus.OUTBOUND_MESSAGE_ACKD
 
-        await work_description.set_status(wd.MessageStatus.OUTBOUND_MESSAGE_ACKD)
+        await work_description.set_outbound_status(wd.MessageStatus.OUTBOUND_MESSAGE_ACKD)
+        persistence.add.assert_called_with(input_data[wd.DATA_KEY], new_data)
+
+    @patch('utilities.timing.get_time')
+    @async_test
+    async def test_set_inbound_status(self, time_mock):
+        time_mock.return_value = '12:00'
+        future = test_utilities.awaitable(old_data)
+
+        persistence = MagicMock()
+        persistence.get.return_value = future
+        persistence.add.return_value = future
+        work_description = wd.WorkDescription(persistence, input_data)
+
+        new_data = copy.deepcopy(input_data)
+        new_data[wd.DATA][wd.INBOUND_STATUS] = wd.MessageStatus.INBOUND_RESPONSE_FAILED
+
+        await work_description.set_inbound_status(wd.MessageStatus.INBOUND_RESPONSE_FAILED)
         persistence.add.assert_called_with(input_data[wd.DATA_KEY], new_data)
 
     def test_null_persistence(self):
         with self.assertRaises(ValueError):
             wd.WorkDescription(None, {'None': 'None'})
 
+    @async_test
+    async def test_update_status(self):
+        new_data = {
+            wd.DATA_KEY: 'aaa-aaa-aaa',
+            wd.DATA: {
+                wd.CREATED_TIMESTAMP: '11:59',
+                wd.LATEST_TIMESTAMP: '13:00',
+                wd.VERSION_KEY: 1,
+                wd.INBOUND_STATUS: None,
+                wd.OUTBOUND_STATUS: wd.MessageStatus.OUTBOUND_MESSAGE_ACKD,
+                wd.WORKFLOW: workflow.SYNC
+            }
+        }
+        persistence = MagicMock()
+        persistence.get.return_value = test_utilities.awaitable(new_data)
+        work_description = wd.WorkDescription(persistence, old_data)
+
+        await work_description.update()
+
+        self.assertEqual(work_description.version, 1)
+        self.assertEqual(work_description.outbound_status, wd.MessageStatus.OUTBOUND_MESSAGE_ACKD)
+        self.assertEqual(work_description.last_modified_timestamp, '13:00')
+
+    @async_test
+    async def test_update_status(self):
```

Comment:
This has the same name as the above method (which I think may result in one of these test cases not running!):
```suggestion
    async def test_update_status_no_data(self):
```